### DISCUSSION TITLE
Ledger Recover

### DISCUSSION THREAD

**Discussion by aido** [aido/app-seed-tool#11] [General]
Ledger Recover is a service offered by Ledger for the backup and restoration of a Ledger device's seed.  When subscribed to Ledger Recover, the secure element encrypts and splits the Secret Recovery Phrase into three fragments. These encrypted fragments are then sent through three independent secure channels to third=party backup providers.

[Phase 3 of Ledger's Open Source roadmap](https://github.com/LedgerHQ/recover-whitepaper/blob/main/open_source_roadmap.png) states that Ledger plan to provide tools that will allow an individual to implement their own shard backup provider. This may or may not make app-sskr-check obsolete.

The [Ledger Recover white paper](https://github.com/LedgerHQ/recover-whitepaper) shows that Ledger will be using a variant of Shamir's Secret Sharing called Pedersen Verifiable Secret Sharing.. My concern here is that Ledger will be the only hardware wallet using this flavour of Pedersen VSS so it may not be interoperable. Other standards already exist but Ledger now seem to have created their own non-interoperable one. One of the reasons for choosing SSKR for the app was because of its interoperability.

I will be slowing down or even pausing development of app-sskr-check until I get a better idea of how Ledger plan to allow individuals implement their own shard backup.

**Comment by InfiniteQE** [comment]
I hope you mean that your project app-sskr-check is also conceptualised in stages, and that you mean to pause only adding future featues like BIP85.

I think there is an important market for open source airgapped devices, such as Seedsigner.

SLIP39 and SSKR can be done airgapped on a tailsOS, a single board computer, an [Ipod touch](https://github.com/blockchaincommons/GordianSeedTool-iOS), or [Coinbase style in a faraday tent](https://www.wired.com/story/coinbase-physical-vault-to-secure-a-virtual-currency/), but empowering non-technical users to shard, verify and restore remaining airgapped using widely available components is a niche market, but one that would gravitate towards Bitcoin. 

Right now there are up to [6 million](https://twitter.com/Ledger/status/1600129670631944192) ledger customers [who maybe feeling like this](https://old.reddit.com/r/ledgerwallet/comments/14xomse/hows_everyone_feeling_about_their_stax_preorder/).

Even if Ledger's Pedersen Verifiable Secret Sharing standard is amazing, it will be kryptonite to all other hardware wallet vendors.

NanoS with millions of units sold can be initialized/restored/wiped fully airgapped on USB power (Unlike TrezorT's SLIP39), with app-sskr-check providing a unique and valuable tool for those who wish to remain offline and airgapped, and may want to [stamp their mnemonic seed or shamir shards into metal](https://github.com/BlockchainCommons/crypto-commons/blob/master/Docs/sskr-cold-storage.md) without ever using a PC or printer.

**Reply by aido** [reply]
Ledger Head of Embedded Software has commented here:

https://www.reddit.com/r/ledgerwallet/comments/14f5x79/comment/joyniwh/

**Reply by InfiniteQE** [reply]
Ledger have sold 6 million "Nano" hardware wallets, of which the NanoS will not support Recover.  I don't know what their breakdown of S / X / SP are, but I would expect the majority sold have been NanoS.

**Comment by aido** [comment]
When Ledger announced that as part of their [Open Source Roadmap](https://github.com/LedgerHQ/recover-whitepaper/blob/main/open_source_roadmap.png) they would release tools to allow an individual implement their own shard backup provider I was concerned that this could overlap with the app-sskr-check application or even make it obsolete.

However, Ledger have released more details on [how to use these tools](https://github.com/LedgerHQ/blue-loader-python/blob/master/ledgerblue/HowToRecover.md) and I can see there is no overlap with app-sskr-check.
So I have decided that I will continue with the development of this app, possibly under a different name going forward; app-seed-tools or app-seed-utils or something similar.
